<!--
  This file will launch four instances of the mmWave sensor node and configure each TI mmWave sensor using a 3D config. This launch file is intended
  to be used to launch the mmWave sensor node on TI's TDA4VM Jacinto Processors, as such a separate command must be used to launch RViz 
  for visualization on a PC. This File also launches mmWaveSync, a node which implements a user selected HW or SW based sync.
 -->

<launch>

  <node pkg="ti_mmwave_rospkg" type="mmWaveSync" name="mmWaveSync" ns="ti_mmwave_sync" output="screen">
    <param name="sync_type" value="HW" />
    <param name="num_sensors" value="4" />
    <param name="pwm_pin" value="32" />
    <param name="pulse_freq" value="10" />
    <param name="duty_cycle" value="0.001" />
    <param name="trigger_delay" value="0" />
    <param name="config_file" value="$(find ti_mmwave_rospkg)/cfg/1843_3d_hw_trig.cfg" />
  </node>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="0" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyACM0"/>
    <arg name="com_data" value="/dev/ttyACM1"/>
  </include>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="1" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyACM2"/>
    <arg name="com_data" value="/dev/ttyACM3"/>
  </include>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="2" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyACM4"/>
    <arg name="com_data" value="/dev/ttyACM5"/>
  </include>

  <include file="$(find ti_mmwave_rospkg)/launch/sync_single_sensor.launch">
    <arg name="device_num" value="3" />
    <arg name="tf_args" value="0 0 0 0 0 0" />
    <arg name="com_user" value="/dev/ttyACM6"/>
    <arg name="com_data" value="/dev/ttyACM7"/>
  </include>
  
</launch>
